src/crimson/osd/osd.cc

```diff
@@ -1402,6 +1403,93 @@ seastar::future<> OSD::send_beacon()
+                  collection_ref, std::move(cleanup)).then([collection_ref] {
+                    return collection_ref->flush_commit().then([](bool) {
+                      return seastar::make_ready_future<>();
+                    });
+                  });
+              });
+          });
+      }).then([=, this]() {
+        auto start = std::chrono::steady_clock::now();
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution<> dis(0, 255);
+        return seastar::do_with(ceph::os::Transaction(), ceph::os::Transaction(),
+          [=, this, &gen, &dis](auto& t, auto& cleanup) {
+            return seastar::do_for_each(boost::make_counting_iterator<int64_t>(0),
+              boost::make_counting_iterator<int64_t>(count / bsize),
+              [=, &gen, &dis, &t, &cleanup](int64_t i) {
+                std::string data(bsize, 'a');
+                ceph::buffer::list bl;
+                bl.append(data);
+                std::string oid_str;
+                uint64_t offset = 0;
+                if (onum && osize) {
+                  oid_str = fmt::format("disk_bw_test_{}", dis(gen) % onum);
+                  offset = (dis(gen) % (osize / bsize)) * bsize;
+                } else {
+                  oid_str = fmt::format("disk_bw_test_{}", i * bsize);
+                }
+                ghobject_t oid(hobject_t(sobject_t(object_t(oid_str), 0)),
+                  ghobject_t::NO_GEN,
+                  shard_id_t::NO_SHARD);
+                t.write(coll_t::meta(), oid, offset, bsize, bl);
+                cleanup.remove(coll_t::meta(), oid);
+                bl.clear();
+                return seastar::make_ready_future<>();
+              }).then([this, &t, &cleanup]() {
+                return store.get_sharded_store().open_collection(coll_t::meta()).then(
+                  [this, &t, &cleanup](auto collection_ref) {
+                    return store.get_sharded_store().do_transaction(
+                      collection_ref, std::move(t)).then([this, &cleanup, collection_ref] {
+                        return store.get_sharded_store().do_transaction(
+                          collection_ref, std::move(cleanup)).then([collection_ref] {
+                            return collection_ref->flush_commit().then([](bool) {
+                              return seastar::make_ready_future<>();
+                            });
+                          });
+                      });
+                  });
+          }).then([start, elapsed]() {
+            seastar::timer<>::time_point end = std::chrono::steady_clock::now();
+            *elapsed = std::chrono::duration<double>(end - start).count();
+            return seastar::make_ready_future<>();
+          }).handle_exception([FNAME, elapsed](auto eptr) {
+            ERROR("error in run_bench: {}", eptr);
+            *elapsed = -1;
+            return seastar::make_ready_future<>();
```

Comment:
`elapsed` should be carried by the returned future - `seastar::make_ready_future<double>(elapsed)`